test/unit/argument_parser/format_parse_validators_test.cpp

```diff
@@ -113,31 +113,31 @@ TEST(validator_test, input_file)
         std::filesystem::path file_in_path;
 
         // option
-        std::filesystem::path path = tmp_name.get_path();
+        std::string path = tmp_name.get_path().string();
         const char * argv[] = {"./argument_parser_test", "-i", path.c_str()};
```

Comment:
In general, can you remove the construction of the string and just do:

```suggestion
         const char * argv[] = {"./argument_parser_test", "-i", path.string().c_str()};
```

This applies to all the changes in this file.

Reply:
No, no. In general path.string() return a temporal string, and after this expresion ended argv will have a dangling pointer from c_str(): in general we need to keep a copy (or change the type of argv).

> (2.1) — char: The encoding is the native ordinary encoding. The method of conversion, if any, is operating system dependent. [Note: For POSIX-based operating systems path::value_type is char so no conversion from char value type arguments or to char value type return values is performed. For Windows-based operating systems, the native ordinary encoding is determined by calling a Windows API function. —end note] 
(2.2) — wchar_t: The encoding is the native wide encoding. The method of conversion is unspecified. [Note: For Windows-based operating systems path::value_type is wchar_t so no conversion from wchar_t value type arguments or to wchar_t value type return values is performed. —end note]

> 29.11.7.4.6 Native format observers [fs.path.native.obs]
1 The string returned by all native format observers is in the native pathname format (29.11.7).
const string_type& native() const noexcept;
2 Returns: The pathname in the native format.
const value_type* c_str() const noexcept;
3 Effects: Equivalent to: return native().c_str();


This my return wchar_t*

Reply:
You're right!
I like how it is now, though we put a space after the `&`:
```cpp 
std::string const & path
```

Reply:
OK. Do I need to rebase again?